### DISCUSSION TITLE
How can I redirect to a different Host?

### DISCUSSION THREAD

**Discussion by OZoneGuy** [actix/actix-web#2795] [Q&A] [answered]
I am planning to use Azure AD for authentication and I need to redirect the user to the login page for microsoft. I can't find any documentation on how to redirect to a different URL with a different host. Is it currently possible in actix 4?

**Comment by robjtede** [comment, marked as answer]
There are redirect responders in actix-web-lab: https://docs.rs/actix-web-lab/0.16/actix_web_lab/web/struct.Redirect.html#method.to

```rust
async fn handler() -> impl Responder {
  Redirect::to("https://duck.com")
}
```

The Redirect struct will be promoted to Actix Web in an upcoming 4.x release.

You can also construct redirect responses yourself using 30x status codes and a Location header.

**Reply by robjtede** [reply]
This has been added to Actix Web: https://docs.rs/actix-web/4.3/actix_web/web/struct.Redirect.html#method.to

**Comment by OZoneGuy** [comment]
Isn't there anything in the `actix_web` package? Do I need the lab?

**Reply by westy92** [reply]
You can now use `actix_web::redirect::Redirect` as of 4.3.

**Reply by KaiyuanMa** [reply]
I got "module `redirect` is private module" When using `use actix_web::redirect::Redirect;`

**Reply by robjtede** [reply]
@KaiyuanMa  See https://docs.rs/actix-web/4.3/actix_web/web/struct.Redirect.html#method.to
